### PR TITLE
Push switch conditions/tasks

### PR DESCRIPTION
**Pull request by lcoderre**

### CODE REVIEW

File: app/models/udevice.js [hunk 1]
```diff
@@ -372,7 +376,7 @@ UDeviceSchema.statics.fromNinjaBlocks = function (ninjaDevice, ninjaDeviceId, ni
 	}
 
 	if (device.type == 1012) {
```
Comment (bobbylavoie) [inline app/models/udevice.js:378]: i already did it in UDevice.fromSpecialCase and UDevice.toSpecialCase.. you can delete this condition.

File: app/models/utask.js [hunk 2]
```diff
@@ -111,16 +111,26 @@ UTaskSchema.statics.toNinjaBlocks = function (task, cb) {
...
+
 	
 	UScenario.findById(task._scenario, function(err, scenario){
 		UDevice.findById(scenario._device, function(err, device){
 			var deviceTpIdSplit = device.tpId.split(":");//subdevice data, if one, is stored into id.
 			ninjaRule.actions[0].params.guid = deviceTpIdSplit[0];
-			ninjaRule.actions[0].params.da = UDevice.toSpecialCase(device.type, task.value);
+
+			// if device is a subdevice
+			if (deviceTpIdSplit.length > 1) {
```
Comment (bobbylavoie) [inline app/models/utask.js:126]: You can use the UDevice.toSpecialCase() function for the switch condition..

**Comment by bobbylavoie** [comment]
Good job mon Wrench

File: app/models/utask.js (hunk 2 above)
Comment (lcoderre) [inline app/models/utask.js:126]: I could indeed return a different data format depending on the type/value, but I'm also using that condition to change the shortname. Not a big deal, let me know if I still should move that code :+1:

**Pull request merged**